atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/progress/SweepProgressStore.java

```diff
@@ -15,42 +15,57 @@
 import com.palantir.atlasdb.keyvalue.api.KeyValueService;
 import com.palantir.atlasdb.keyvalue.api.RangeRequest;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
 import com.palantir.atlasdb.schema.generated.SweepProgressTable;
-import com.palantir.atlasdb.schema.generated.SweepProgressTable.SweepProgressRow;
-import com.palantir.atlasdb.schema.generated.SweepProgressTable.SweepProgressRowResult;
 import com.palantir.atlasdb.schema.generated.SweepTableFactory;
-import com.palantir.atlasdb.transaction.api.Transaction;
 
 public class SweepProgressStore {
+    private static final Logger log = LoggerFactory.getLogger(SweepProgressStore.class);
 
     private final KeyValueService kvs;
-    private final SweepTableFactory tableFactory;
+    private final SweepProgressTable progressTable;
+    private final TableReference tableRef;
+
+    private static final byte[] row = PtBytes.toBytes("dummy");
+    private static final byte[] column = PtBytes.toBytes("s");
+    private static final Cell CELL = Cell.create(row, column);
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module());
 
     public SweepProgressStore(KeyValueService kvs, SweepTableFactory tableFactory) {
         this.kvs = kvs;
-        this.tableFactory = tableFactory;
+        this.progressTable = tableFactory.getSweepProgressTable(null);
+        this.tableRef = progressTable.getTableRef();
     }
 
-    public Optional<SweepProgress> loadProgress(Transaction tx)  {
-        SweepProgressTable progressTable = tableFactory.getSweepProgressTable(tx);
-        Optional<SweepProgressRowResult> result = Optional.ofNullable(
-                progressTable.getRow(SweepProgressRow.of(0)).orElse(null));
-        return result.map(SweepProgressStore::hydrateProgress);
+    public Optional<SweepProgress> loadProgress()  {
+        Map<Cell, Value> entry = kvs.get(tableRef, ImmutableMap.of(CELL, Long.MAX_VALUE));
+        return hydrateProgress(entry);
     }
 
-    public void saveProgress(Transaction tx, SweepProgress progress) {
-        SweepProgressTable progressTable = tableFactory.getSweepProgressTable(tx);
-        SweepProgressRow row = SweepProgressRow.of(0);
-        progressTable.putFullTableName(row, progress.tableRef().getQualifiedName());
-        progressTable.putStartRow(row, progress.startRow());
-        progressTable.putCellsDeleted(row, progress.staleValuesDeleted());
-        progressTable.putCellsExamined(row, progress.cellTsPairsExamined());
-        progressTable.putMinimumSweptTimestamp(row, progress.minimumSweptTimestamp());
+    public void saveProgress(SweepProgress newProgress) {
+        Optional<SweepProgress> oldProgress = loadProgress();
+        try {
+            kvs.checkAndSet(casProgressRequest(oldProgress, newProgress));
+        } catch (Exception e) {
+            log.warn("Exception trying to persist sweep progress.", e);
+        }
```

Comment:
Let's discuss in person. I'm somehow not certain that we need to use CAS - it seems like `put` would be sufficient, since I'm not sure we really care about the old value. If there are multiple active sweepers this could get interesting, but using CAS doesn't seem to handle that case well either.

Reply:
Note that the sweep lock (distinct from the backup lock) should prevent us from having multiple concurrent sweepers in normal usage. We should verify that this would also stop the sweep CLI running while a background sweep was in progress (and whether the sweep CLI writes to progress).